realm/realm-annotations-processor/src/main/java/io/realm/processor/RealmVersionChecker.java

```diff
@@ -70,11 +71,35 @@ public void run() {
     private void launchRealmCheck() {
         //Check Realm version server
         String latestVersionStr = checkLatestVersion();
-        if (!latestVersionStr.equals(REALM_VERSION)) {
+        if (reportNewVersion(latestVersionStr, REALM_VERSION)) {
             printMessage("Version " + latestVersionStr + " of Realm is now available: " + REALM_ANDROID_DOWNLOAD_URL);
         }
     }
 
+    /**
+     * Checks if the latest release should be reported to the user.
+     *
+     * Note: Public for testing.
+     *
+     * @param newVersion new version string.
+     * @param currentVersion current version string.
+     * @return returns {@code true} if the new version detected should be reported to the user, {@code false} if not.
+     */
+
+    public static boolean reportNewVersion(String newVersion, String currentVersion) {
+        if (newVersion == null || newVersion.equals("")) {
+            // Invalid version strings should be ignored
+            return false;
+        }
+        if (newVersion.toLowerCase(Locale.US).endsWith("beta")) {
```

Comment:
Are you planning only beta version for Realm 4.0? Or also alpha/RC or anything else?
Why not checking if there is a prefix (`-something`) after the version number?